requires.py

```diff
@@ -12,139 +12,183 @@
 # limitations under the License.
 import json
 
-from charms.reactive import RelationBase
-from charms.reactive import hook
-from charms.reactive import scopes
+from charms.reactive import (
+    Endpoint,
+    set_flag,
+    clear_flag
+)
 
-from charmhelpers.core import hookenv
+from charms.reactive import (
+    when,
+    when_any
+)
 
+from charmhelpers.core.hookenv import (
+    log,
+    related_units
+)
 
-class KubeControlRequireer(RelationBase):
-    """Implements the kubernetes-worker side of the kube-control interface.
 
-    """
-    scope = scopes.GLOBAL
 
-    @hook('{requires:kube-control}-relation-{joined,changed}')
+class KubeControlRequirer(Endpoint):
+    """
+    Implements the kubernetes-worker side of the kube-control interface.
+    """
+    @when_any('endpoint.{endpoint_name}.joined',
+              'endpoint.{endpoint_name}.changed')
     def joined_or_changed(self):
-        """Set states corresponding to the data we have.
-
         """
-        conv = self.conversation()
-        conv.set_state('{relation_name}.connected')
+        Set states corresponding to the data we have.
+        """
+        set_flag(self.expand_name('{endpoint_name}.available'))
         self.check_states()
 
-    @hook('{requires:kube-control}-relation-departed')
+    @when('endpoint.{endpoint_name}.departed')
```

Comment:
This flag actually isn't cleared automatically.  This handler should clear it, I think.